x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1303,6 +1307,70 @@ class AgentPolicyService {
     }));
   }
 
+  public async turnOffAgentTamperProtections(soClient: SavedObjectsClientContract): Promise<{
+    updatedPolicies: Array<Partial<AgentPolicy>> | null;
+    failedPolicies: Array<{ id: string; error: Error | SavedObjectError }>;
+  }> {
+    const { saved_objects: agentPoliciesWithEnabledAgentTamperProtection } =
+      await soClient.find<AgentPolicySOAttributes>({
+        type: SAVED_OBJECT_TYPE,
+        page: 1,
+        perPage: SO_SEARCH_LIMIT,
+        filter: normalizeKuery(SAVED_OBJECT_TYPE, 'ingest-agent-policies.is_protected: true'),
+        fields: ['revision'],
+      });
+
```

Comment:
Question: Can we leave it at `SO_SEARCH_LIMIT`?

Reply:
I think that's fine since only the `revision` field is queried. 
Though I wonder if the solution should support more than 10k agent policies.